engine/include/Engine/Intermediate/DataChunk.inl

```diff
@@ -0,0 +1,52 @@
+#include "Engine/Intermediate/DataChunk.hpp"
+
+template <typename TStoredComponent, int TSize>
+constexpr inline DataChunk<TStoredComponent, TSize>::DataChunk() noexcept
+{
+    m_components.reserve(TSize / sizeof(TStoredComponent));
+}
+
+template <typename TStoredComponent, int TSize>
+template <typename... Args>
+TStoredComponent& DataChunk<TStoredComponent, TSize>::addComponent(Args&&... args) noexcept
+{
+    std::unique_lock lock(m_mutex);
+
+    return m_components.emplace_back(std::forward<Args>(args)...);
+}
+
+template <typename TStoredComponent, int TSize>
+void DataChunk<TStoredComponent, TSize>::destroyComponent(const TStoredComponent* componentToDestroy) noexcept
+{
+    std::unique_lock lock(m_mutex);
+
+    for (size_t i = 0; i < m_components.size(); ++i)
+    {
+        if (unlikely(&m_components[i] == componentToDestroy))
+        {
+            std::swap(m_components[i], m_components.back());
+
+            m_components[i].getGameObject().updateComponentLink<TStoredComponent>(&m_components.back(),
+                                                                                  &m_components[i]);
+            m_components.pop_back();
+        }
+    }
+}
+
+template <typename TStoredComponent, int TSize>
+DataChunk<TStoredComponent, TSize>* DataChunk<TStoredComponent, TSize>::getInstance() noexcept
+{
+    // double same if to avoid to lock mutex
+    if (unlikely(m_pInstance == nullptr))
```

Comment:
WARNING : Can cause a data race if m_pInstance is not atomic.
We can do different things : 
- put m_pInstance atomic
- remove the double check

We could also make an init() function that will instance m_pInstance, and use the mutex only in here.